elephant/signal_processing.py

```diff
@@ -1,17 +1,136 @@
-# -*- coding: utf-8 -*-
-"""
-docstring goes here.
+'''
+signal_processing module
 
-:copyright: Copyright 2015 by the Elephant team, see AUTHORS.txt.
+:copyright: Copyright 2014 by the Elephant team, see AUTHORS.txt.
 :license: Modified BSD, see LICENSE.txt for details.
-"""
+'''
 
+from __future__ import division, print_function
 import numpy as np
 import scipy.signal
 import quantities as pq
 import neo
 
 
+def zscore(signal, inplace=True):
+    '''
+    Apply a z-score operation to one or several AnalogSignalArray objects.
+
+    The z-score operation subtracts the mean :math:`\\mu` of the signal, and
+    divides by its standard deviation :math:`\\sigma`:
+
+    .. math::
+         Z(x(t))= \\frac{x(t)-\\mu}{\\sigma}
+
+    If an AnalogSignalArray containing multiple signals is provided, the
+    z-transform is always calculated for each signal individually.
+
+    If a list of AnalogSignalArray objects is supplied, the mean and standard
+    deviation are calculated across all objects of the list. Thus, all list
+    elements are z-transformed by the same values of :math:`\\mu` and
+    :math:`\\sigma`. For AnalogSignalArrays, each signal of the array is
+    treated separately across list elements. Therefore, the number of signals
+    must be identical for each AnalogSignalArray of the list.
+
+    Parameters
+    ----------
+    signal : neo.AnalogSignalArray or list of neo.AnalogSignalArray
+        Signals for which to calculate the z-score.
+    inplace : bool
+        If True, the contents of the input signal(s) is replaced by the
+        z-transformed signal. Otherwise, a copy of the original
+        AnalogSignalArray(s) is returned. Default: True
+
+    Returns
+    -------
+    neo.AnalogSignalArray or list of neo.AnalogSignalArray
+        The output format matches the input format: for each supplied
+        AnalogSignalArray object a corresponding object is returned containing
+        the z-transformed signal with the unit dimensionless.
+
+    Use Case
+    --------
+    You may supply a list of AnalogSignalArray objects, where each object in
+    the list contains the data of one trial of the experiment, and each signal
+    of the AnalogSignalArray corresponds to the recordings from one specific
+    electrode in a particular trial. In this scenario, you will z-transform the
+    signal of each electrode separately, but transform all trials of a given
+    electrode in the same way.
+
+    Example
+    -------
+    >>> a=neo.AnalogSignalArray(
```

Comment:
This is not a valid way to create an AnalogSignalArray (Neo should complain, but doesn't) - it should always be 2D.